### PR TITLE
Add RegionMap.from_dataframe() and RegionMap.as_dict() methods

### PR DESCRIPTION
**Pull request by adrien-berchet**
I need to create custom hierarchies and the nested dict structure is very boring to update so I would prefer updating the DataFrame representation and then be able to create a RegionMap object from it.

I just need a clarification: in the tests the root node ID is `-1`, which means that in the DataFrame representation it is its own parent since the root parent ID is always `-1`. I feel it weird so I changed the test to ensure that no node have a `-1` ID. But I can change if you think that this situation actually makes sense, and I could consider that the first node with `parent_id == -1` is the root node for example.

### CODE REVIEW

File: voxcell/region_map.py [hunk 1]
```diff
@@ -142,6 +142,38 @@ def as_dataframe(self):
         ret.loc[:, 'children_count'] = [len(self._children[_id]) for _id in ret.index.to_list()]
         return ret
 
+    @classmethod
+    def from_dataframe(cls, hierarchy_df):
+        """Converts a DataFrame to a region_map.
+
+        Note: the 'root' node should have a parent value of -1.
+        """
+        nodes = hierarchy_df.to_dict(orient="index")
```
Comment (eleftherioszisis) [inline voxcell/region_map.py:151]: Does this need to be sorted so that we always have the same hierarchy with the same nodes, or it doesn't really matter?

**Comment by eleftherioszisis** [comment]
Trying it with the full hierarchy:
```python
from voxcell import RegionMap
r = RegionMap.load_json("mba_hierarchy.json")
df = r.as_dataframe()
r2 = RegionMap.from_dataframe(df)
should_be_None = r2.get(997, attr="parent_structure_id")
type(should_be_None)
>> float
```
Returns a float NaN which does not pass checks like X is None and is not consistent with the initial region map's None value.

**Comment by adrien-berchet** [comment]
> I'm afraid it's hard to know when we should convert `nan` back to `None`, except if we know the list of columns for which the cast should be performed.

Of maybe we can consider that we should convert `nan` to `None` for all columns with `dtype=float`?

File: voxcell/region_map.py (hunk 1 above)
Comment (eleftherioszisis) [reply voxcell/region_map.py:155]: Yeah, I guess so.

File: voxcell/region_map.py [hunk 2]
```diff
@@ -142,6 +143,57 @@ def as_dataframe(self):
...
+        for k, v in nodes.items():
+            v["id"] = k
+            v.pop("children_count", None)
+            parent_id = v.pop("parent_id", None)
+            for float_col in float_cols:
+                if float_col in v:
+                    if np.isnan(v[float_col]):
+                        v[float_col] = None
+                    elif float_col in float_int_cols:
+                        v[float_col] = int(v[float_col])
```
Comment (eleftherioszisis) [inline voxcell/region_map.py:176]: Could you add tests with a more complex hierarchy that covers this? It seems that the dummy hierarchy doesn't have all the different data columns that are present in a real hierarchy and the real issues behind the conversion is left out.

Comment (adrien-berchet) [reply voxcell/region_map.py:176]: I did with a RegionMap loaded from `tests/data/region_map.json` which contains more complex data that fails without this new change. And there are both `None` and `int` values so it covers both cases.

Comment (adrien-berchet) [reply voxcell/region_map.py:176]: The only thing that is not covered id the case with multiple nodes with `parent_id == -1`. I can add a test for this.

**Comment by eleftherioszisis** [comment]
LGTM. Maybe @mgeplf would also like to take a look.

**Pull request merged**